### PR TITLE
editoast: fix: diesel max params on big infra

### LINKED ISSUE #2301: Refactor persist batch functions in editoast
https://github.com/DGEXSolutions/osrd/blob/1bea0353a54b1b76a52bed05ee2bf6c999848c87/editoast/src/schema/track_section.rs#L36-L57

This functions is duplicated for each object type.

- [ ] Implement a generic functions
- [ ] Adapt callers

### LINKED ISSUE #2328: Import railjson command fails with big infra
**Describe the bug**
When trying to import an infra with more than $2^{16} - 1$ objects (of the same type), the command fails.

**To Reproduce**
1. In editoast run this command: `editoast import-railjson my_infra /path/to/infra.json`

Make sure that the infra you're importing is big enough.
The command fails with the following error message:

> An internal diesel error occurred: 'number of parameters must be between 0 and 65535

**Expected behavior**

The command should work.

**How to fix**:

The functions `persist_batch` must handle vectors with a lot of elements by doing multiple pg queries.

### PR DESCRIPTION
**Pull request by Khoyo**
Diesel only accepts 65535 parameters per query, so we need to chunk our insertions when importing big infras.

Fixes #2328.
Fixes #2301.

### CODE REVIEW

**Comment by Khoyo** [comment]
TODO: 

- [x] A test
- [x] We need to refactor these methods, they pretty much do 11 time the same thing

File: editoast/src/schema/buffer_stop.rs [hunk 1]
```diff
@@ -43,9 +43,11 @@ impl BufferStop {
             })
             .collect::<Vec<_>>();
 
-        diesel::insert_into(osrd_infra_bufferstopmodel)
-            .values(datas)
-            .execute(conn)?;
+        for data_chunk in datas.chunks(65534) {
```
Comment (flomonster) [inline editoast/src/schema/buffer_stop.rs]: - Why your chunk size equals $2^{16} - 2$ and not $2^{16} - 1$?
- You could have a const variable that contains the value `2_usize.pow(16) - 1`

Comment (Khoyo) [reply editoast/src/schema/buffer_stop.rs]: Done and done :)

**Comment by Khoyo** [comment]
About the refactor:

Why not a generic function? Mostly because diesel goes insane when you try to write it. If someone is able too, please do.

Why not raw SQL? Because AFAIK we don't have a good way to escape SQL in diesel (basically we must call the unsafe fn `PQescapeLiteral()`), and we'd loose the compile time type safety provided by diesel. Still, I'm open to it if people think a macro is too big a gun, I know I can reach for them to fast :)

Codecov is complaining, but does it really make sense to move the test into the macro or duplicate it 11 times ?

File: editoast/editoast_derive/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                        .values(data_chunk)
+                        .execute(conn)?;
+                }
+
+                Ok(())
+            }
+        }
+    };
+
+    proc_macro::TokenStream::from(expanded)
```
Comment (flomonster) [inline editoast/editoast_derive/src/lib.rs:64]:
```suggestion
    expended.into()
```

File: editoast/src/schema/railjson.rs [hunk 3]
```diff
@@ -71,6 +71,7 @@ impl RailJson {
             Detector::persist_batch(&self.detectors, infra.id, conn)?;
             OperationalPoint::persist_batch(&self.operational_points, infra.id, conn)?;
             Route::persist_batch(&self.routes, infra.id, conn)?;
+
```
Comment (flomonster) [inline editoast/src/schema/railjson.rs:74]: I think this was commited by mistake.

**Review by flomonster — CHANGES_REQUESTED** [review]
Great PR!

**Review by flomonster — DISMISSED** [review]

**Pull request merged**